optimade/validator/utils.py

```diff
@@ -199,8 +202,19 @@ def __init__(
         self.response = None
         self.max_retries = max_retries
         self.headers = headers or {}
+        if "User-Agent" not in self.headers:
+            self.headers["User-Agent"] = DEFAULT_USER_AGENT_STRING
         self.timeout = timeout or DEFAULT_CONN_TIMEOUT
         self.read_timeout = read_timeout or DEFAULT_READ_TIMEOUT
+        self.httpx_timeout = httpx.Timeout(self.timeout, read=self.read_timeout)
+        self.http_client = httpx.Client()
+
+    def close(self):
+        """Close the underlying httpx client, if it is still open."""
+        try:
+            self.http_client.close()
+        except Exception:
```

Comment:
Isn't there an explicit exception for this to use instead, so that unexpected exceptions are not missed?

Reply:
I was in two minds here really, the exceptions raised by `close()` are not documented so I don't think we can necessarily rely on them. The method already has handling for closing already closed connections (it just skips) so I think this is safe enough.

Reply:
(see https://github.com/encode/httpx/blob/1c33a2854e5d18955c8766c5c01e6229716b5b7a/httpx/_client.py#L1249-L1259)

Reply:
To me this seems more like you should simply call `self.http_client.close()` and that's that? If an exception is raised it's something to be concerned about, if not, then it's already closed or it was successfully closed.